internal/output/release/release.go

```diff
@@ -0,0 +1,235 @@
+  fi
+
+  git checkout $2
+  git fetch
+  git rebase upstream/$2
+  git cherry-pick -x $1
+}
+
+function commit {
+  if [ $# -ne 1 ]; then
+    echo 1>&2 "Usage: $0 commit <tag>"
+    exit 1
+  fi
+
+  git commit -s -m "release($1): prepare release" -m "This is the official $1 release."
+}
+
+if declare -f "$1" > /dev/null
+then
+  cmd="$1"
+  shift
+  $cmd "$@"
+else
+  cat <<EOF
+Usage:
+  commit:       Create the official release commit message.
+  cherry-pick:  Cherry-pick a commit into a release branch.
+  changelog:    Update the specified CHANGELOG.
+EOF
+
+  exit 1
+fi`
+
+const configStr = `style: github
+template: CHANGELOG.tpl.md
+info:
+  title: CHANGELOG
+  repository_url: https://github.com/talos-systems/talos
+options:
+  commits:
+    # filters:
+    #   Type:
+    #     - feat
+    #     - fix
+    #     - perf
+    #     - refactor
+  commit_groups:
+    # title_maps:
+    #   feat: Features
+    #   fix: Bug Fixes
+    #   perf: Performance Improvements
+    #   refactor: Code Refactoring
+  header:
+    pattern: "^(\\w*)(?:\\(([\\w\\$\\.\\-\\*\\s]*)\\))?\\:\\s(.*)$"
+    pattern_maps:
+      - Type
+      - Scope
+      - Subject
+  notes:
+    keywords:
+      - BREAKING CHANGE`
+
+const templateStr = `{{ range .Versions }}
+<a name="{{ .Tag.Name }}"></a>
+## {{ if .Tag.Previous }}[{{ .Tag.Name }}]({{ $.Info.RepositoryURL }}/compare/{{ .Tag.Previous.Name }}...{{ .Tag.Name }}){{ else }}{{ .Tag.Name }}{{ end }} ({{ datetime "2006-01-02" .Tag.Date }})
+
+{{ range .CommitGroups -}}
+### {{ .Title }}
+
+{{ range .Commits -}}
+* {{ if .Scope }}**{{ .Scope }}:** {{ end }}{{ .Subject }}
+{{ end }}
+{{ end -}}
+
+{{- if .NoteGroups -}}
+{{ range .NoteGroups -}}
+### {{ .Title }}
+
+{{ range .Notes }}
+{{ .Body }}
+{{ end }}
+{{ end -}}
+{{ end -}}
+{{ end -}}`
+
+// Output implements .gitignore generation.
+type Output struct {
+	output.FileAdapter
+}
+
+// NewOutput creates new Makefile output.
+func NewOutput() *Output {
+	output := &Output{}
+
+	output.FileAdapter.FileWriter = output
+
+	return output
+}
+
+// Compile implements output.Writer interface.
+func (o *Output) Compile(node interface{}) error {
+	compiler, implements := node.(Compiler)
+
+	if !implements {
+		return nil
+	}
+
+	return compiler.CompileRelease(o)
+}
+
+// Filenames implements output.FileWriter interface.
+func (o *Output) Filenames() []string {
+	return []string{release, config, template}
+}
+
+// GenerateFile implements output.FileWriter interface.
+func (o *Output) GenerateFile(filename string, w io.Writer) error {
+	type writer func(io.Writer) error
+
+	var f writer
+
+	switch filename {
+	case release:
+		f = o.release
+	case config:
+		f = o.config
+	case template:
+		f = o.template
+	default:
+		panic("unexpected filename: " + filename)
+	}
+
+	if err := os.MkdirAll(filepath.Dir(filename), 0o755); err != nil {
+		return err
+	}
+
+	if err := f(w); err != nil {
+		return err
+	}
+
+	if filename == release {
+		return os.Chmod(filename, 0o744)
```

Comment:
I'm not sure what's the best way in general here, but at the moment `GenerateFile` is called file might not exist yet (it's buffered for content comparison).

Probably same way, provide some interface to return default file mode on creation (optional interface?)

Reply:
Yeah, I wasn't sure what the best way was. I will dig into it some more.

Reply:
Done.